core/nwb.base.yaml

```diff
@@ -64,11 +64,21 @@ datasets:
 
 - neurodata_type_def: ExternalImage
   neurodata_type_inc: BaseImage
-  doc: A type for referencing an external image file. The single file path or URL to the external image file should
-    be stored in the dataset. This type should NOT be used if the image is stored 
-     in another NWB file and that file is linked to this file.
+  doc: A type for referencing an external image file. The single file path or URI to the
+    external image file should be stored in the dataset. This type should NOT be used if
+    the image is stored in another NWB file and that file is linked to this file.
   dtype: text
   # shape: scalar  # this will be supported in the NWB schema language 2.0
+  attributes:
+  - name: image_mode
+    dtype: text
+    doc: Image mode (color mode) of the image, e.g., "RGB", "RGBA", "grayscale", and "LA".
+    required: true
+  - name: image_format
+    dtype: text
+    doc: Common name of the image file format. Only widely readable, open file formats are allowed.
+      Allowed values are "PNG", "JPEG", "GIF", and "TIFF".
```

Comment:
webp did not convince you?
https://developers.google.com/speed/webp

Reply:
I did not find easy ways to open webp images in python and matlab. Maybe that's OK but to me, that's a sign that the images are not widely readable. I get that these external images will most likely be stimuli presented to the user so maybe they don't need to be readable in common NWB programming languages. Have you encountered real-life experimental data that uses webp to store images?

Reply:
Doesn't pillow reads webp?

https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#webp

The matlab limitation seems like a good argument against though.

> Have you encountered real-life experimental data that uses webp to store images?
I have not. I think I added in the discussion because @yarikoptic mentioned it and it seems the format has some advantages in terms of compression but the criteria of widespread available seems like a good one to me.

Reply:
Ah. I missed that Pillow can read webp. Thanks.

We can always add webp later

Reply:
Agree.